src/java.base/share/classes/java/lang/Float.java

```diff
@@ -1097,6 +1098,7 @@ public static short floatToFloat16(float f) {
             // Preserve sign and attempt to preserve significand bits
             return (short)(sign_bit
                     | 0x7c00 // max exponent + 1
+                    | 0x0200 // QNaN
```

Comment:
I don't understand what is being done here. From IEEE 754-2019:

"Besides issues such as byte order which affect all data
interchange, certain implementation options allowed by this standard must also be considered:
― for binary formats, how signaling NaNs are distinguished from quiet NaNs
― for decimal formats, whether binary or decimal encoding is used.
This standard does not define how these parameters are to be communicated."

The code in java.lang.Float in particular is meant to be usable on all host CPUs so architecture-specific assumptions about QNAN vs SNAN should be avoided.